specs/activity-api.yaml

```diff
@@ -515,43 +441,273 @@ components:
+            }
+        },
+        {
+            "run": {
+                "entry_point": "rust-wasi-tutorial",
+                "args": ["/input/file_in", "/output/file_cp"]
+            }
+        },
+        ...
+        ]
+        ```
+
+        ### Rust
+
+        For Rust - this format is a default representation of Rust enum types, as serialized by `serde` library.
+        Therefore it is recommended to use this yaml specification to manually specify Rust enum types rather than depend on
+        automatically-generated code.
+
+    '
+      allOf:
+        - $ref: '#/components/schemas/Object'
+        - type: object
+
+    DeployCommand:
+      allOf:
+        - $ref: '#/components/schemas/ExeScriptCommand'
+        - type: object
+      properties:
+        deploy:
+          type: object
+
+    StartCommand:
+      allOf:
+        - $ref: '#/components/schemas/ExeScriptCommand'
+        - type: object
+      properties:
+        start:
+          $ref: '#/components/schemas/StartCommandBody'
+
+    StartCommandBody:
+      type: object
+      properties:
+        args:
+            type: array
+            items:
+              type: string
+
+    TransferCommand:
+      allOf:
+        - $ref: '#/components/schemas/ExeScriptCommand'
+        - type: object
+      properties:
+        transfer:
+          $ref: '#/components/schemas/TransferCommandBody'
+
+    TransferCommandBody:
+      type: object
+      properties:
+        from:
+          type: string
+        to:
+          type: string
+        format:
+          type: string
+        depth:
+          type: number
+        fileset:
+          type: array
+          items:
+            $ref: '#/components/schemas/FileSet'
+
+    FileSet:
+      properties:
+        desc:
+          type: string
+        includes:
+          type: array
+          items:
+            type: string
+        excludes:
+          type: array
+          items:
+            type: string
+
+    RunCommand:
+      allOf:
+        - $ref: '#/components/schemas/ExeScriptCommand'
+        - type: object
+      properties:
+        run:
+          $ref: '#/components/schemas/RunCommandBody'
+
+    RunCommandBody:
+      type: object
+      properties:
+        entry_point:
+          type: string
+        args:
+          type: array
+          items:
+            type: string
+
+    # End ExeScript structures
+
+    ExeScriptCommandState:
+      required:
+        - batchId
+        - command
+      type: object
+      properties:
+        batchId:
+          type: string
+        command:
+          type: string
+        progress:
+          type: string
+        params:
+          type: array
+          items:
+            type: string
+
     ExeScriptCommandResult:
       required:
         - index
+        - eventDate
         - result
       type: object
       properties:
         index:
           type: integer
           format: int32
           minimum: 0
+        eventDate:
```

Comment:
no longer needed

Reply:
is it really? We may not be filtering by timestamp, but the timestamp itself (ie moment in time when the command has completed on provider node) seems relevant information.

Reply:
ok, adding it for future use